packages/shared/src/telemetry/throttler.ts

```diff
@@ -0,0 +1,104 @@
+import type { TelemetryEventRaw } from './types';
+
+type TtlInMilliseconds = number;
+
+const DEFAULT_CACHE_TTL_MS = 86400000; // 24 hours
+
+/**
+ * Manages throttling for telemetry events using the browser's localStorage to
+ * mitigate event flooding in frequently executed code paths.
+ */
+export class TelemetryEventThrottler {
+  #storageKey = 'clerk_telemetry';
+  #cacheTtl = DEFAULT_CACHE_TTL_MS;
+
+  isEventThrottled(event: TelemetryEventRaw): boolean {
+    if (this.#isValidBrowser) {
+      return false;
+    }
+
+    const now = Date.now();
+    const key = this.#generateKey(event);
+    const entry = this.#cache?.[key];
+
+    if (!entry) {
+      const updatedCache = {
+        ...this.#cache,
+        [key]: now,
+      };
+
+      localStorage.setItem(this.#storageKey, JSON.stringify(updatedCache));
+    }
+
+    const shouldInvalidate = entry && now - entry > this.#cacheTtl;
+    if (shouldInvalidate) {
+      const updatedCache = this.#cache;
+      delete updatedCache[key];
+
+      localStorage.setItem(this.#storageKey, JSON.stringify(updatedCache));
+    }
+
+    return !!entry;
+  }
+
+  /**
+   * Generates a consistent unique key for telemetry events by sorting payload properties.
+   * This ensures that payloads with identical content in different orders produce the same key.
+   */
+  #generateKey({ event, payload }: TelemetryEventRaw): string {
```

Comment:
Just need to refactor this according to @BRKalow feedback, we should use the `preparedPayload` rather than the raw one.